pkg/graph/resolvers/mto_solution.go

```diff
@@ -81,27 +78,24 @@ func MTOSolutionCreateCustom(
 }
 
 func MTOSolutionCreateCommon(
+	ctx context.Context,
 	logger *zap.Logger,
 	principal authentication.Principal,
 	store *storage.Store,
 	modelPlanID uuid.UUID,
-	commonSolutionKey *models.MTOCommonSolutionKey,
+	commonSolutionKey models.MTOCommonSolutionKey,
 ) (*models.MTOSolution, error) {
 	principalAccount := principal.Account()
 	if principalAccount == nil {
 		return nil, fmt.Errorf("principal doesn't have an account, username %s", principal.String())
 	}
-
 	// TODO: Populate data from mto common solution
 	mtoSolution := models.NewMTOSolution(
 		modelPlanID,
-		commonSolutionKey,
-		nil,
+		&commonSolutionKey,
 		nil,
-		models.MTOFacilitatorOther,
+		nil, // type is not stored for a solution from the common library
```

Comment:
Similar comments on the other `nil` fields would be helpful, sometimes reading `nil, nil, nil` can be a bit confusing

Reply:
That's a good call, thanks Clay. I added it. I removed the TODO as well, we don't actually need anything else from the common solution table at this point.